src/providers/npm.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Callable, Tuple
+import requests
+import urllib.request
+import os
+import multiprocessing
+import itertools
+from semver import max_satisfying
+
+from src.providers.Provider import Provider
+
+NPM_REGISTRY_URL = 'https://registry.npmjs.org/'
+
+
+def _clean_package_version(version: str):
+    if version == '*' or version is None or version == 'latest':
+        return 'latest'
+    else:
+        return version.replace('~', '').replace('^', '')
+
+
+def _get_version_package_payload(package_name: str, version: str) -> dict:
+    response = requests.get(NPM_REGISTRY_URL + package_name)
+    if response.status_code == 404:
+        raise Exception('Module not found {0}:{1}'.format(package_name, version))
+    if response.status_code != 200:
+        raise Exception('Unknown error occurred {0}:{1}'.format(package_name, version))
+    response_payload = response.json()
+    if version == 'latest':
+        version = response_payload['dist-tags']['latest']
+    if version not in response_payload['versions']:
+        satisfied_version = max_satisfying(list(response_payload['versions'].keys()), version, loose=False)
+        if version is None:
+            print('Failed to satisfy {0}:{1}'.format(package_name, version))
```

Comment:
This log seems redundant... The exception should be enough, no?

Reply:
I agree

Reply:
done